swagger_to/style.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+"""
+Generates code for an Elm client.
+"""
+# pylint: disable=too-many-lines
+from typing import MutableSet, MutableMapping, List
+
+import swagger_to.intermediate
+import swagger_to.swagger
+import swagger_to
+
+
+def check_header(swagger: swagger_to.swagger) -> List[str]:
+    """
+    Checks whether the swagger header conforms to the style guide.
+
+    :param swagger: parsed swagger file
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+
+    if swagger.name != swagger_to.snake_case(swagger.name):
+        errs.append("Swagger name {} is not snake case.".format(swagger.name))
+
+    if not swagger.base_path.startswith("/"):
+        errs.append("Swagger base path {} doesn't start with a slash.".format(swagger.base_path))
+
+    if swagger.description.capitalize() != swagger.description:
+        errs.append("Swagger description \"{}\" should be capitalized")
+
+    return errs
+
+
+def check_cases(typedefs: MutableMapping[str, swagger_to.intermediate.Typedef],
+                endpoints: List[swagger_to.intermediate.Endpoint]) -> List[str]:
+    """
+    Checks whether the typedefs and endpoints conform to the casing conventions.
+
+    :param param typedefs: swagger type definitions
+    :param endpoints: swagger endpoint definitions
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+
+    seen_typedefs = set()  # type: MutableSet
+    for _, typedef in enumerate(typedefs.values()):
+        errs.extend(_check_recursively_cases(typedef=typedef, visited=seen_typedefs))
+
+    for endpoint in endpoints:
+        if endpoint.path != swagger_to.snake_case(endpoint.path):
+            errs.append("Path {} must conform to snake case.".format(endpoint.path))
+        if endpoint.operation_id != swagger_to.snake_case(endpoint.operation_id):
+            errs.append("{} is not a snake case identifier.".format(endpoint.operation_id))
+        for param in endpoint.parameters:
+            if param.name != swagger_to.snake_case(param.name):
+                errs.append("In endpoint {}: parameter {} has not a snake case identifier.".format(
+                    endpoint.path, param.name))
+
+    return errs
+
+
+def check_descriptions(typedefs: MutableMapping[str, swagger_to.intermediate.Typedef],
```

Comment:
Same as above, I'd split this in two (`check_typedef_descriptions`, `check_endpoint_descriptions`) for readability.